documentation/modules/ROOT/pages/configuration/storage.adoc

```diff
@@ -0,0 +1,705 @@
+|No default
+|The password that will be used to access Redis.
+|[[schema-history-internal-redis-db-index]]<<schema-history-internal-redis-db-index, `schema.history.internal.redis.db.index`>>
+|0
+|The database index (0..15) that will be used to access Redis.
+|[[schema-history-internal-redis-ssl-enabled]]<<schema-history-internal-redis-ssl-enabled,`schema.history.internal.storage.redis.ssl.enabled`>>
+|false
+|Use SSL for Redis connection.
+|[[schema-history-internal-redis-connection-timeout-ms]]<<schema-history-internal-redis-connection-timeout-ms, `schema.history.internal.storage.redis.connection.timeout.ms`>>
+|2000
+|Connection timeout (in ms).
+|[[schema-history-internal-redis-socket-timeout-ms]]<<schema-history-internal-redis-socket-timeout-ms, `schema.history.internal.storage.redis.socket.timeout.ms`>>
+|2000
+|Socket timeout (in ms).
+|[[schema-history-internal-redis-retry-initial-delay-ms]]<<schema-history-internal-redis-retry-initial-delay-ms, `schema.history.internal.storage.redis.retry.initial.delay.ms`>>
+|300
+|Initial retry delay (in ms).
+|[[schema-history-internal-redis-retry-max-delay-ms]]<<schema-history-internal-redis-retry-max-delay-ms, `schema.history.internal.storage.redis.retry.max.delay.ms`>>
+|10000
+|Maximum retry delay (in ms).
+|[[schema-history-internal-redis-retry-max-attempts]]<<schema-history-internal-redis-retry-max-attempts, `schema.history.internal.storage.redis.retry.max.attempts`>>
+|10
+|Maximum number of retry attempts before giving up.
+|[[schema-history-internal-redis-wait-enabled]]<<schema-history-internal-redis-wait-enabled,`schema.history.internal.storage.redis.wait.enabled`>>
+|false
+|Enables wait for replica. In case Redis is configured with a replica shard, this allows to verify that the data has been written to the replica.
+|[[schema-history-internal-redis-wait-timeout-ms]]<<schema-history-internal-redis-wait-timeout-ms,`schema.history.internal.storage.redis.wait.timeout.ms`>>
+|1000
+|Timeout when wait for replica.
+|[[schema-history-internal-redis-wait-retry-enabled]]<<schema-history-internal-redis-wait-retry-enabled, `schema.history.internal.storage.redis.wait.retry.enabled`>>
+|false
+|Enables retry on wait for replica failure.
+|[[schema-history-internal-redis-wait-retry-delay-ms]]<<schema-history-internal-redis-wait-retry-delay-ms, `schema.history.internal.storage.redis.wait.retry.delay.ms`>>
+|1000
+|Delay of retry on wait for replica failure.
+
+|===
+
+== Amazon S3
+The store uses link:https://aws.amazon.com/s3/[Amazon S3] object storage service.
+It is typically used when Debezium is deployed in link:https://aws.amazon.com/msk/[Amazon Managed Streaming] service.
+
+=== Internal schema history store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[s3-history-type]]<<s3-history-type, `schema.history.internal`>>
+|No default
+|Must be set to `io.debezium.storage.s3.history.S3SchemaHistory`.
```

Comment:
If this and similar properties have required values, can they be coded to carry the appropriate defaults?